lang/cs/Org.Apache.REEF.Network/Group/Pipelining/Impl/DoubleArrayPipelineDataConverter.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using Org.Apache.REEF.Network.Group.Config;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+
+namespace Org.Apache.REEF.Network.Group.Pipelining.Impl
+{
+    public sealed class DoubleArrayPipelineDataConverter : IPipelineDataConverter<double[]>
+    {
+        private readonly int _pipelineMessageSize;
+        private static readonly Logger Logger = Logger.GetLogger(typeof(DoubleArrayPipelineDataConverter));
+
+        [Inject]
+        private DoubleArrayPipelineDataConverter([Parameter(typeof(GroupCommConfigurationOptions.PipelineMessageSize))] int pipelineMessageSize)
+        {
+            if (pipelineMessageSize <= 0)
+            {
+                throw new ArgumentException("PipelinePackageElements must be strictly positive");
+            }
+            _pipelineMessageSize = pipelineMessageSize;
+        }
+
+        public List<PipelineMessage<double[]>> PipelineMessage(double[] message)
```

Comment:
100% agree here. This is something I have been thinking about too. I think the next step should be to see if we can refactor these all out to use one of the interfaces you suggest. I currently am not sure how embedded `List` is in our current implementation, but I don't think it's going to be too tight, as we don't really use it to its full capacity yet.

Would you be okay putting that in a separate pull request?